core/src/main/scala/org/apache/spark/sql/delta/stats/StatisticsCollection.scala

```diff
@@ -389,7 +384,306 @@ trait StatisticsCollection extends DeltaLogging {
+   * Note: This method is matching the logical name of tables with the columns inside
+   * DeltaConfigs.DATA_SKIPPING_STATS_COLUMNS.
+   */
+  def renameDeltaStatsColumn(
+      metadata: Metadata,
+      oldColumnPath: Seq[String],
+      newColumnPath: Seq[String]): Map[String, String] = {
+    if (oldColumnPath == newColumnPath) return Map.empty[String, String]
+    val deltaStatsColumnSpec = configuredDeltaStatsColumnSpec(metadata)
+    deltaStatsColumnSpec.deltaStatsColumnNamesOpt.map { deltaColumnsNames =>
+      val deltaStatsColumnsPath = deltaColumnsNames
+        .map(_.nameParts)
+        .map { attributeNameParts =>
+          val commonPrefix = oldColumnPath.zip(attributeNameParts)
+            .takeWhile { case (left, right) => left == right }
+            .size
+          if (commonPrefix == oldColumnPath.size) {
+            newColumnPath ++ attributeNameParts.takeRight(attributeNameParts.size - commonPrefix)
+          } else {
+            attributeNameParts
+          }
+        }
+        .map(columnParts => UnresolvedAttribute(columnParts).name)
+      Map(
+        DeltaConfigs.DATA_SKIPPING_STATS_COLUMNS.key -> deltaStatsColumnsPath.mkString(",")
+      )
+    }.getOrElse(Map.empty[String, String])
+  }
+
+  /** Returns the configured set of columns to be used for stats collection on a table */
+  def configuredDeltaStatsColumnSpec(metadata: Metadata): DeltaStatsColumnSpec = {
+    val indexedColNamesOpt = DeltaConfigs.DATA_SKIPPING_STATS_COLUMNS.fromMetaData(metadata)
+    val numIndexedCols = DeltaConfigs.DATA_SKIPPING_NUM_INDEXED_COLS.fromMetaData(metadata)
+    indexedColNamesOpt.map { indexedColNames =>
+      DeltaStatsColumnSpec(parseDeltaStatsColumnNames(indexedColNames), None)
+    }.getOrElse {
+      DeltaStatsColumnSpec(None, Some(numIndexedCols))
+    }
+  }
+
+  /**
+   * Generates a filtered data schema for stats collection.
+   * Note: This method is matching the logical name of tables with the columns inside
+   * DeltaConfigs.DATA_SKIPPING_STATS_COLUMNS. The output of the filter schema is translated into
+   * physical name.
+   *
+   * @param schemaNames the full name path of all columns inside `schema`.
+   * @param schema the original data schema.
+   * @param statsColPaths the specific set of columns to collect stats on.
+   * @param mappingMode the column mapping mode of this statistics collection.
+   * @param parentPath the parent column path of `schema`.
+   * @return filtered schema
+   */
+  def filterSchema(
+      schemaNames: Seq[String],
+      schema: StructType,
+      statsColPaths: Seq[Seq[String]],
+      mappingMode: DeltaColumnMappingMode,
+      parentPath: Seq[String] = Seq.empty): Option[StructType] = {
+    // Find the unique column names at this nesting depth, each with its path remainders (if any)
+    val cols = statsColPaths.groupBy(_.head).mapValues(_.map(_.tail))
+    val newSchema = schema.flatMap { field =>
+      cols.get(field.name).flatMap { paths =>
```

Comment:
so if the field name in the property is "x.y.z" and `x` is the `field` here.. then `paths` is `[y, z]` right?

Reply:
if the field name in the property is `x.y.z`, then `z` is the tail field.
`x.y` is the path to `z`.

This is an example 
```
CREATE TABLE T1 (c0 long, c1 STRUCT <c11: String, c12: long, c13 STRUCT <c131: long, c132: long>>) using delta
TBLPROPERTIES('delta.dataSkippingStatsColumns' = 'c0, c1.c11, c1.c12, c1.c13.c131, c1.c13.c132');
```